XdmodAID.md

```diff
@@ -0,0 +1,868 @@
+focused on in getting data in on the OpenStack side and how I found the flavors that
+were deleted, I was surprised that it wasn't noticed by anyone until we were gathering
+data.  So I started looking in to this.
+
+With kim, we looked at the database structure and the data that was being pulled from
+OpenShift.  The type coming from openshift is in fractions of a CPU which is a floating
+point type, where as the database field that this is stored in is an iteger.  Addtionoally,
+in the supremm database, this becomes a floating point.
+
+I have just started to work with the data that is processed here as it was not inserting any
+CPU data into the jobs table.  This was due to the data coming from openshift often in
+fractional amounts of CPU.  One simple solution to this is to store the data in milli CPU by
+multiplying by 1000.  This was tested in the xdmod-staging project on the nerc-infra cluster
+and seemed to work (ie, I got milli CPU in the jobs table).
+
+I then tried to setup Supremm on the staging project, and the process failed when running
+from xdmod-setup. I did get the staging mariadb setup by restoring from a backup of production
+though something is wrong with the configuration of mongo
+
+
+
+# xdmod-hierarchy
+
+In order to report to insititutions, departments and PIs; a system wide hierarchy can
+be defined to aggreate statistics at the appropriate level.  This is done using
+csv files that define each level - the xdmod documentation  explaination can be
+found in (https://open.xdmod.org/10.0/hierarchy.html,
+https://open.xdmod.org/10.0/user-names.html, https://open.xdmod.org/10.0/cloud.html).
+I will try to cosolidate and explain this a bit better.
+
+Depending on if it is a HPC job or a cloud resource there are different levels to
+the hierarchy.  This is due to jobs not having a project name associated with
+them.  To side by side compare:
+```
+        Job(OpenShift)              cloud(OpenStack)             file
+        -----------------------------------------------------------------------
+        Unit                        Unit                         hierarchy.csv
+          division                    division                   hierarchy.csv
+             department                 department               hierarchy.csv
+               PI                         PI                     group.csv
+                 group name                 group name           names.csv
+                                              project name       pi2project.csv
+```
+Since we have OpenShift implemented as an HPC resoource and openstack as a cloud
+resource, we need to effectively remove a layer of the cloud hierarchy.  This was
+done by mapping OpenStack projects to OpenStack projects in the pi2project.csv
+file.
+
+Defining the first 3 levels of the hierarchy is done via the hierarchy.json file
+just as explained in the xdmod documentation.
+
+Unfortunately, when developing this used different names to define the PI
```

Comment:
used -> xdmod used
or maybe instead of xdmod, I? I am not sure of the desired info here.